### ISSUE TITLE
Request: VYV Photon control

### ISSUE THREAD

**Issue by matthewcraigtaylor** [bitfocus/companion#165]
Photon, has a similar feature set as D3. One way it beats D3 is the inbuilt cue(macro) system. Anything in the software can be recorded as a macro and executed as a cue. As such, when we externalise control, the simple way is to recall a cue within photon: it will do the rest. The manual lists many more parameters that can be externalised, but with the well implemented cue system I feel it is surplus to requirement. 

Companion is a wonderful piece of software to maximise the use of an inexpensive box. Thank you for the work you have put in.

**Comment by willosof** [comment]
Sure, we can make photon support. Do you have any documentation for the tcp/udp control? :)

**Comment by matthewcraigtaylor** [comment]
Yes I do, however I have not checked the most up to date info as their portal hasn't been working for me today.
https://www.vyv.ca/

From my notes however:
`<photon> CUE_EXEC 2 </photon>`

Send on TCP or UDP to port 55503 

The number “2” is the ID number displayed in Cue Editor.

Cheers

**Comment by willosof** [comment]
Anything else than recalling cues you want in?

**Comment by matthewcraigtaylor** [comment]
No thank you.  By using the cues/macros it gives complete control and ability to edit actions and timings from within the software.

**Comment by PerRoine** [comment]
Module is made Waiting for repo and link to push module needs to be tested.

**Comment by PerRoine** [comment]
Module is made needs testing to see if it Works

**Comment by matthewcraigtaylor** [comment]
Fantastic, that's fast work.

**Comment by matthewcraigtaylor** [comment]
I have tested this module and it is not quite right.
I then created a python script to prove the system worked, then tested companion again, I am therefore certain the system is good but the module is not.  I will paste the python script below.

Thank you
![python send](https://user-images.githubusercontent.com/34844462/45970125-9f17d280-c02d-11e8-9755-efa943c9a029.PNG)

**Comment by haakonnessjoen** [comment]
The module in companion uses TCP, since it was said that it supports both UDP and TCP. Could you do a test with python using TCP?

**Comment by matthewcraigtaylor** [comment]
[Photon_UDP_control_v7.8.pdf](https://github.com/bitfocus/companion/files/2416256/Photon_UDP_control_v7.8.pdf)

**Comment by matthewcraigtaylor** [comment]
Haakonnessjoen, I (due to my limited knowledge) have not been able to get TCP to work from Python despite the information in the manual.
Thank you

**Comment by matthewcraigtaylor** [comment]
Hello,
I have had an opportunity to talk to VYV Photon and do some more testing.
The results are in:
The following command needs to be sent if using TCP, although UDP also accepts the same format.

![image](https://user-images.githubusercontent.com/34844462/49016087-f0068d00-f17c-11e8-8aa6-1cfa6bafedc0.png)


We have also tested UDP vs. TCP.  UDP is much faster so that would be the preference for Companion. 
Thank you

**Comment by matthewcraigtaylor** [comment]
Hello,

Is there any update on this one?

Thank you.
Matthew
